tests/integration/RDBTestCase.php

```diff
@@ -0,0 +1,174 @@
+<?php declare(strict_types = 1);
+
+use RemoteDataBlocks\Config\DataSource\HttpDataSource;
+use RemoteDataBlocks\Config\Query\HttpQuery;
+use RemoteDataBlocks\Config\Query\HttpQueryInterface;
+use RemoteDataBlocks\Config\QueryRunner\QueryRunner;
+use RemoteDataBlocks\Editor\BlockManagement\BlockRegistration;
+use RemoteDataBlocks\Editor\BlockManagement\ConfigStore;
+
+class RDBTestCase extends WP_UnitTestCase {
+	// Query mocking
+
+	/**
+	 * @psalm-suppress PossiblyUnusedParam
```

Comment:
Psalm decided [every method parameter in this class](https://github.com/Automattic/remote-data-blocks/actions/runs/12957874218/job/36147086879) was "possibly unused" for some reason I wasn't able to determine, so I used these to quiet the failures.

Reply:
Weird, I "fixed" it by using a regular PHPUnit `TestCase` instead of `WP_UnitTestCase`. Guessing Psalm somehow needs to know about `WP_UnitTestCase` via some stubs or something?

At some point we'll probably need the factory methods that `WP_UnitTestCase` class provides, but ok for now.

Reply:
I'm actually surprised this still works! `WP_UnitTestCase` does all sorts of stuff (in [`WP_UnitTestCase`](https://github.com/WordPress/wordpress-develop/blob/a82874058f58575dbba64ce09b6dcbd43ccf5fdc/tests/phpunit/includes/testcase.php) -> [`WP_UnitTestCase_Base`](https://github.com/WordPress/wordpress-develop/blob/a82874058f58575dbba64ce09b6dcbd43ccf5fdc/tests/phpunit/includes/abstract-testcase.php)) beyond factory methods that actually seem useful for integration tests, like a lot of [test cleanup](https://github.com/WordPress/wordpress-develop/blob/a82874058f58575dbba64ce09b6dcbd43ccf5fdc/tests/phpunit/includes/abstract-testcase.php#L84-L97). I'm going to look further into this to see if I can add whatever stubs we need to keep static analysis happy.

Reply:
@chriszarate Managed to make psalm happy with `WP_UnitTestCase` using [`preloadClasses="true"`](https://psalm.dev/docs/running_psalm/configuration/#stubs) in the stubs section. I'm honestly not sure why this fixes `PossiblyUnusedParam`, but the docs mention abstract classes (which `WP_UnitTestCase` is) and it works now. Weird!